lib/src/widgets/chat/typing/ds_dot_animation.widget.dart

```diff
@@ -0,0 +1,150 @@
+import 'package:flutter/material.dart';
+
+import 'ds_dot.widget.dart';
+
+/// Creates a list of animated dots for use in typing identification and other situations
+///
+/// To define the stitch appearance use [size] and [color]
+/// Use [numberOfDots] to set the amount of dots and use [innerPadding] to set the spacing between them
+/// To change the animation speed use [animationTime]
+/// To set the dot ascent level use [upLevelAnimation]
+class DSDotAnimation extends StatefulWidget {
+  final double size;
+  final Color color;
+  final int numberOfDots;
+  final double innerPadding;
+  final Duration animationTime;
+  final double upLevelAnimation;
+
+  const DSDotAnimation({
+    Key? key,
+    this.numberOfDots = 3,
+    this.size = 10,
+    this.innerPadding = 2.5,
+    this.animationTime = const Duration(milliseconds: 160),
+    this.color = Colors.black,
+    this.upLevelAnimation = -9,
+  }) : super(key: key);
+
+  @override
+  State<DSDotAnimation> createState() => _DSDotAnimationState();
+}
+
+class _DSDotAnimationState extends State<DSDotAnimation>
+    with TickerProviderStateMixin {
+  List<AnimationController>? _animationControllers;
+
+  final List<Animation<double>> _animations = [];
+
+  @override
+  void initState() {
+    super.initState();
+    _initAnimation();
+  }
+
+  @override
+  void dispose() {
+    for (var controller in _animationControllers!) {
+      controller.dispose();
+    }
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Directionality(
+      textDirection: TextDirection.ltr,
+      child: Center(
+        child: Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: List.generate(
+            widget.numberOfDots,
+            (index) {
+              return DotAnimated(
+                animationControllers: _animationControllers,
+                widget: widget,
+                animations: _animations,
+                index: index,
+              );
+            },
+          ).toList(),
+        ),
+      ),
+    );
+  }
+
+  /// Controls the animation of the dots
+  void _initAnimation() {
+    _animationControllers = List.generate(
+      widget.numberOfDots,
+      (index) {
+        return AnimationController(
+          vsync: this,
+          duration: widget.animationTime,
+        );
+      },
+    ).toList();
+
+    for (int i = 0; i < widget.numberOfDots; i++) {
+      _animations.add(Tween<double>(begin: 0, end: widget.upLevelAnimation)
```

Comment:
The begin and end values should match the prototype animation. Maybe these values should do the trick:

```dart
Tween<double>(begin: 4, end: -3,)
```